app/controllers/accounts_controller.rb

```diff
@@ -39,8 +40,8 @@ def show
       end
 
       format.json do
-        expires_in 3.minutes, public: true
-        render json: @account, content_type: 'application/activity+json', serializer: ActivityPub::ActorSerializer, adapter: ActivityPub::Adapter
+        expires_in 3.minutes, public: !(authorized_fetch_mode? && signed_request_account.present?)
```

Comment:
we should add `Vary: Signature` to the response too right? Otherwise you might get served a `public` response for a signed request. Do you know the best way to do that? Might be best to just make this `!authorized_fetch_mode?`

Reply:
I did do that and then @BenLubar told me I should do this. If the cache header says "private" it's not supposed to be shared anyway..?

Reply:
Right, but if the sequence of requests goes like this:

- client A requests the account without authentication and gets the limited account response (public: true)
- The intermediary cache caches this response publicly
- client B requests the account *with* authentication and the cache responds with the limited account response.

then that's bad

Reply:
Okay, Vary header added.